python/metricq/data_client.py

```diff
@@ -102,7 +102,8 @@ def __init__(self, *args, **kwargs):
             await self.data_channel.close()
             self.data_channel = None
         if self.data_connection:
-            await self.data_connection.close(exception)
+            # We need not pass anything as exception to this close. It will only hurt.
```

Comment:
Sure, but why will it hurt? Does passing on the exception lead to trouble further down the stack?

Reply:
It does in fact hurt very much, it was the source of the actual issue. Normally, when no exception is passed, `aiormq` uses `asyncio.CancelledError`, which leads to a normal quiet shutdown. If we pass `None`, however, the default from `aiormq.Connection.close` is overwritten and `aiormq.FutureStore.reject_all` will use an `Exception` instead creating all kind of noise / errors within the loop. I consider this kinda weird behavior within `aiormq`. Nevertheless, if we receive an actual Exception here, I don't see the point of passing it to all tasks/futures that are related to the connection. I think this will just lead to duplication of the exception, which we pass to super().stop for proper reporting anyway.

Reply:
That makes sense and would explain some of the erratic behavior (exceptions appear to race themselves on shutdown) I sometimes encountered. But I never figured out a way to actually reproduce these issues. Now I know,
![image](https://user-images.githubusercontent.com/10487782/77197900-46bc7680-6ae6-11ea-96df-ae2f7bb37c26.png)